packages/trie/src/trie/trie.ts

```diff
@@ -376,16 +376,17 @@ export class Trie {
     const nodeKey = this.root()
     const progress: Nibbles = []
     const remaining = bytesToNibbles(key)
+    let node: TrieNode | null = null
     try {
-      await this.processNode(nodeKey, stack, progress, remaining)
+      node = await this.processNode(nodeKey, stack, progress, remaining)
     } catch (error: any) {
       if (error.message === 'Missing node in DB' && !throwIfMissing) {
         // pass
       } else {
         throw error
       }
     }
-    return { node: null, stack, remaining }
+    return { node, stack, remaining }
```

Comment:
At the end I thought that we might be able to skip this `node` return value since the node returned is just the last item on the stack (if the stack is complete), so this information is just redundant. I generally had the impression that this could be significantly simplified (so: the return format), not sure e.g., is `remaining` necessary or used? Might be though.

Reply:
you're probably right.